### PR TITLE
fix: fix default pools and refactor

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
default pools doesn't work in slurm because they don't always use configs. This refactors the code to use the resource manager resource pool functions and modularizes parts of the api handlers.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/api_resourcepool.go [hunk 1]
```diff
@@ -203,6 +170,43 @@ func (a *apiServer) ListWorkspacesBoundToRP(
...
+		sproto.GetDefaultAuxResourcePoolRequest{})
+	if err != nil {
+		return err
+	}
+
+	if poolName == defaultComputePool.PoolName ||
+		poolName == defaultAuxPool.PoolName {
+		return errors.Errorf("default resource pool %s cannot be bound to any workspace",
+			poolName)
+	}
```
Comment (stoksc) [inline master/internal/api_resourcepool.go:190]:
```suggestion
	isDefaultCompute := poolName == defaultComputePool.PoolName
	isDefaultAux := poolName == defaultAuxPool.PoolName
	if isDefaultCompute || isDefaultAux {
		return fmt.Errorf(
			"default resource pool %s cannot be bound to any workspace",
			poolName,
		)
	}
```

File: master/internal/api_resourcepool.go [hunk 2]
```diff
@@ -203,6 +170,43 @@ func (a *apiServer) ListWorkspacesBoundToRP(
...
 
+func (a *apiServer) checkIfPoolIsDefault(poolName string) error {
+	defaultComputePool, err := a.m.rm.GetDefaultComputeResourcePool(a.m.system,
+		sproto.GetDefaultComputeResourcePoolRequest{})
+	if err != nil {
+		return err
+	}
+
+	defaultAuxPool, err := a.m.rm.GetDefaultAuxResourcePool(a.m.system,
+		sproto.GetDefaultAuxResourcePoolRequest{})
```
Comment (stoksc) [inline master/internal/api_resourcepool.go:181]:
```suggestion
	defaultAuxPool, err := a.m.rm.GetDefaultAuxResourcePool(
		a.m.system,
		sproto.GetDefaultAuxResourcePoolRequest{},
	)
```

File: master/internal/api_resourcepool.go [hunk 3]
```diff
@@ -203,6 +170,43 @@ func (a *apiServer) ListWorkspacesBoundToRP(
...
+}
+
+func (a *apiServer) canUserModifyWorkspaces(ctx context.Context, ids []int32) error {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return err
+	}
+
+	if err = workspaceauth.AuthZProvider.Get().CanModifyRPWorkspaceBindings(ctx, *curUser,
+		ids); err != nil {
```
Comment (stoksc) [inline master/internal/api_resourcepool.go:201]: lots of indent nits here. i would really just like if we could remove the line length constraint, but that is a battle for another day.

File: master/internal/api_resourcepool.go [hunk 4]
```diff
@@ -203,6 +170,43 @@ func (a *apiServer) ListWorkspacesBoundToRP(
...
+		return authz.SubIfUnauthorized(err,
+			errors.Errorf(
```
Comment (stoksc) [inline master/internal/api_resourcepool.go:203]:
```suggestion
			fmt.Errorf(
```

**Review by stoksc — APPROVED** [review]
lgtm, just style nits (though def change the errors -> fmt stuff)

**Review by jgongd — APPROVED** [review]
looks good!

**Pull request merged**